toolchain/BUILD.bazel

```diff
@@ -1,47 +1,75 @@
-load("@bazel_skylib//rules:common_settings.bzl", "bool_setting")
+load("@bazel_skylib//rules:common_settings.bzl", "bool_setting", "string_setting", "string_flag")
 load("@rules_cc//cc/toolchains:args.bzl", "cc_args")
 load("@rules_cc//cc/toolchains:artifacts.bzl", "cc_artifact_name_pattern")
 load("@rules_cc//cc/toolchains/impl:documented_api.bzl", "cc_args_list")
 load(":declare_toolchains.bzl", "declare_toolchains")
 
-bool_setting(
-    name = "bootstrap_setting",
-    build_setting_default = False,
+# TODO(cerisier): Find a better name ?
+# This should be the semantic for "what are we building ?"
+# Runtimes without dependencies, runtimes with dependencies, user programs with runtime dependencies...
+string_setting(
+    name = "runtime_stage",
+    values = [
+        "stage0", # depend on compiler-only: CRTs, builtins, libc, static libraries.
+        # "stage1", # depends on above: libunwind
+        # "stage2", # depends on above: libc++abi
+        # "stage3", # depends on above: libc++
+        "complete", # depends on above: user programs.
```

Comment:
I added those as stages because later, we will need to break dependency cycles for instance when building `libc++.so` which needs to depend on `libc.so, crt and compiler-rt`.